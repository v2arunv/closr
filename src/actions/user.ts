import {IUser} from "../models/users";

export interface IUserAction {
    type: string,
    user: IUser,
}

export const setUser = (user: IUser): IUserAction => {
    return <IUserAction> {
        type: 'SET_USER',
        user: user,
    }
};