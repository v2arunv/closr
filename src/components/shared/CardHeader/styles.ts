import {StyleSheet} from "react-native";
import common from '../../../common/styles';

const styles = StyleSheet.create({
    container: {
        paddingTop: 10,
        paddingBottom: 10,
        minWidth: 100,
        paddingLeft: 10,
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        borderBottomColor: common.borderColor,
        borderBottomWidth: 0.5,
    },
    pictureContainer: {
        borderRadius: 10
    },
    picture: {
        height: 50,
        width: 50,
        borderRadius: 25,
    },
    nameContainer: {
        marginLeft: 10,
    },
    name: {
        fontWeight: 'bold'
    }
});

export default styles