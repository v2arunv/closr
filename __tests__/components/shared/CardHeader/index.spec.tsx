import React from 'react';
import CardHeader from '@sharedComponents/CardHeader';
import renderer from 'react-test-renderer';
import user from '@mocks/users';

test('renders correctly', () => {
    const tree = renderer.create(
        <CardHeader
            name={'Full Name'}
            dpURL={'http://supposedToBeALinkToAPicture.com/somepicture.jpeg'}
            onClick={() => {}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});