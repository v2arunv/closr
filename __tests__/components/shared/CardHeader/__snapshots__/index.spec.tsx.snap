// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#dfdfdf",
      "borderBottomWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
      "minWidth": 100,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 10,
        "opacity": 1,
      }
    }
  >
    <Image
      defaultSource={
        Object {
          "testUri": "../../../src/assets/images/dp_placeholder.jpg",
        }
      }
      source={
        Object {
          "uri": "http://supposedToBeALinkToAPicture.com/somepicture.jpeg",
        }
      }
      style={
        Object {
          "borderRadius": 25,
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Full Name
    </Text>
  </View>
</View>
`;
